clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -117,6 +137,341 @@ class image_mem_wrapper {
   image_mem_wrapper *_sub_wrappers{nullptr};
 };
 
+#ifdef _WIN32
+std::unordered_set<ID3D11Resource *> uniqueD3D11Res;
+#endif
+
+/// The wrapper class of graphics interop memory handle.
+class interop_mem_wrapper {
+public:
+#ifdef _WIN32
+  interop_mem_wrapper(ID3D11Resource *d3d11_res, unsigned reg_flags)
+      : _d3d11_res(d3d11_res), _reg_flags(reg_flags) {
+
+    assert(uniqueD3D11Res.find(_d3d11_res) == uniqueD3D11Res.end() &&
```

Comment:
Why doing the check here?

Reply:
To prevent the resource from registering to multiple wrappers.
Similar to the behavior [here](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__D3D11.html#group__CUDART__D3D11_1g85d07753780643584b8febab0370623b:~:text=If%20pD3DResource%20is%20of%20incorrect%20type%20or%20is%20already%20registered%2C%20then%20cudaErrorInvalidResourceHandle%20is%20returned)